src/components/CarInfo.component.tsx

```diff
@@ -0,0 +1,160 @@
+import { useMutation } from "@apollo/client";
+import React, { useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import styled from "styled-components";
+
+import { UserContext } from "../contexts/UserContext";
+import { User_Cars } from "../graphql/generated/graphql";
+import { CREATE_USER_CAR, DELETE_USER_CAR } from "../graphql/mutations";
+import { deleteCarVariables } from "../graphql/variables";
+import { CarRowInfo } from "../interfaces/Car";
+import { defaultTheme } from "../theme";
+import { PATHNAME } from "../utils";
+import { FavoriteButton } from "./FavoriteButton.component";
+import { FlexColumn, FlexRow, H4, P } from "./styled";
+
+export const CarInfo = ({ img, car }: { img: string; car: CarRowInfo }) => {
+  const { loggedUser } = useContext(UserContext);
+  const [isFav, setIsFavorite] = useState<boolean>(car.isFavorite ?? false);
+  const [createUserCar, { loading: favLoading }] = useMutation(CREATE_USER_CAR);
+  const [deleteUserCar, { loading: unfavLoading }] =
+    useMutation(DELETE_USER_CAR);
+
+  const now = new Date();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setIsFavorite(!!car.isFavorite);
+  }, [car.isFavorite]);
```

Comment:
`car.isFavorite` could be undefined (not anymore) because I didn't initialized it on other functions that used the same interface, now I have initialized those functions with `car.isFavorite` set to `false` so this issue have been fixed, I also removed **Nullish coalescing operator (??)**. Thanks again for the feedback Donovan!

Reply:
Also deleted that **useEffect**, and used the `car.isFavorite` instead of the state, it looks like it's working quite good by now.